hs/src/Reach/Linearize.hs

```diff
@@ -338,15 +338,23 @@ data DFEnv = DFEnv
   , eFVMm :: Maybe FVMap
   , eFVE :: FluidEnv
   , eFVs :: [FluidVar]
+  , eBals :: Integer
   }
 
 df_allocVar :: DFApp Int
 df_allocVar = allocVar eCounter_df
 
+adjustFluids :: FluidVar -> DFApp FluidVar
+adjustFluids = \case
+  FV_balances _ _ -> do
+    eBals <- asks eBals
+    return $ FV_balances eBals mempty
```

Comment:
I temporarily did all the work of tweaking the balance values/types to the correct array size in Linearize just to see it work. I think I want to make a new pass immediately after evaluation